### PR TITLE
Cleanup of bugs and better error handling in `user delete`.

### PR DESCRIPTION
**Pull request by tylercloke**
See full commit message for cbc0a26 to see motivation behind deleting (bugged) `--no-disassociate-user` flag.

### CODE REVIEW

**Comment by tylercloke** [comment]
Also open to leaving the option in and WARNing that it doesn't actually do anything and marking as deprecated if we are worried this might break someone's automation...

File: lib/chef/knife/opc_user_delete.rb [hunk 1]
```diff
@@ -20,25 +20,36 @@
...
+      org_names = orgs.map {|o| o['organization']['name']}
+      org_names.each do |org|
+        begin
+          ui.output root_rest.delete("organizations/#{org}/users/#{username}")
+        rescue Net::HTTPServerException => e
+          body = Chef::JSONCompat.from_json(e.response.body)
+          if e.response.code == "403" && body["error"] == "Please remove #{username} from this organization's admins group before removing him or her from the organization."
+            ui.error "Error removing user #{username} from org #{org} due to user being in that org's admins group."
+            ui.msg "Please remove #{username} from the admins group for #{org} before deleting the user."
+            ui.msg "This can be accomplished by passing --force to the org user remove command."
```
Comment (stevendanna) [inline lib/chef/knife/opc_user_delete.rb]: Perhaps give the actual command to run here?

**Comment by stevendanna** [comment]
> Also open to leaving the option in and WARNing that it doesn't actually do anything and marking as deprecated if we are worried this might break someone's automation...

That might be a good addition.  Overall though I am :+1:, this is another thing that the API should probably just enforce itself.  I'm a bit surprised it doesn't.

File: lib/chef/knife/opc_user_delete.rb (hunk 1 above)
Comment (tylercloke) [inline lib/chef/knife/opc_user_delete.rb]: This is bleg, but the reason I didn't is the main place this is used now-a-days is in `chef-server-ctl` and the wrappers aren't a direct map (well they almost are). So if I said `run knife opc org user remove #{org} #{user} --force` that would be confusing csc users, as it is just `chef-server-ctl org-remove-user #{org} #{user} --force` for them. Open to opinions on the right thing to do, but it's just part of life with the current state of our mediocre wrapping.

Comment (tylercloke) [inline lib/chef/knife/opc_user_delete.rb]: Long term, once server-admins is complete and we move this and acl into core knife, I would like to just replace all the wrapping with `chef-server-ctl knife` and `chef-server-ctl knife-org #{org}` that just wraps Chef Server embedded knife and passes things along.

Comment (stevendanna) [inline lib/chef/knife/opc_user_delete.rb]: Ahh, that makes sense.

**Comment by tas50** [comment]
@tylercloke @stevendanna Any reason to keep this open?

**Comment by lamont-granquist** [comment]
knife-opc has been merged into chef-client https://github.com/chef/chef/pull/10187

**Pull request closed without merging**